planner/core/find_best_task.go

```diff
@@ -1239,6 +1289,27 @@ func (ds *DataSource) convertToPartialTableScan(prop *property.PhysicalProperty,
 		tablePlan.SetChildren(ts)
 		return tablePlan
 	}
+	if matchProp {
+		ts.KeepOrder = true
+		ts.Desc = prop.SortItems[0].Desc
+		if ts.Table.GetPartitionInfo() != nil && ts.SCtx().GetSessionVars().StmtCtx.UseDynamicPartitionPrune() {
+			ts.Columns = append(ts.Columns, model.NewExtraPhysTblIDColInfo())
+			ts.schema.Append(&expression.Column{
+				RetType:  types.NewFieldType(mysql.TypeLonglong),
+				UniqueID: ds.ctx.GetSessionVars().AllocPlanColumnID(),
+				ID:       model.ExtraPhysTblID,
+			})
+		}
+		// Add sort items for index scan for merge-sort operation between partitions.
+		byItems := make([]*util.ByItems, 0, len(prop.SortItems))
+		for _, si := range prop.SortItems {
+			byItems = append(byItems, &util.ByItems{
+				Expr: si.Col,
+				Desc: si.Desc,
+			})
+		}
+		ts.ByItems = byItems
```

Comment:
Looks like we can pass `[]*util.ByItems` to this function to reduce the repeated code here and in `convertToPartialIndexScan`.

Reply:
PTAL again. The `byItems` for each partial plan are the same.
Do we need to build new `byItems` for each of them? It looks like sharing the same one is enough.